clash-prelude/src/Clash/Class/BitPack/Internal.hs

```diff
@@ -12,7 +12,7 @@ Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
 {-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE UndecidableInstances #-}
-
+{-# LANGUAGE PolyKinds #-} -- Required for BitPack (KnownNat n) => BitPack (Proxy n) instances
```

Comment:
Sorry I missed this on initial review, but what does `BitPack (KnownNat n)` mean? I initially mentally autocorrected this to `KnownNat n => BitPack (Proxy n)` which makes sense although it is overly specific: it forces `n` to be a `Nat` which needs `PolyKinds`. How about:

```suggestion
{-# LANGUAGE PolyKinds #-} -- Required for BitPack (Proxy (n :: Nat)) instances
```

It's still overly specific, but at least it makes sense. I don't think a `BitPack (KnownNat n)` constraint means anything?